optimizationmodule/Function.go

```diff
@@ -0,0 +1,53 @@
+package optimizationmodule
+
+import (
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"slices"
+)
+
+type Function interface {
+	Call(wrapper *openrtb_ext.RequestWrapper) (string, error)
+}
+
+// will be used in Build rules trie
+func NewFunction(name string, params []string) Function {
+	switch name {
```

Comment:
We'll want to make this case insensitive. Recommend converting to lowercase and having all the cases expressed in lowercase too.